Sources/FoundationEssentials/String/String+IO.swift

```diff
@@ -184,6 +189,17 @@ extension String {
                 }
             }
             self = bytes.withContiguousStorageIfAvailable(buildString) ?? Array(bytes).withUnsafeBufferPointer(buildString)
+        case .japaneseEUC:
+            // Here we catch encodings that are supported by Foundation Framework
+            // but are not supported by corelibs-foundation.
+            // We delegate conversion to ICU.
+            guard let string = (
+                bytes.withContiguousStorageIfAvailable({ _icuMakeStringFromBytes($0, encoding: encoding) }) ??
```

Comment:
Actually sorry one more thing that I just caught - this `_icuMakeStringFromBytes` function is only defined in `!FOUNDATION_FRAMEWORK` builds so this fails to build with `FOUNDATION_FRAMEWORK` enabled. Could you wrap this case in `#if !FOUNDATION_FRAMEWORK`. For now in Foundation.framework this will fall back to `NSString` below, and we can look at using ICU instead of `NSString` in Foundation.framework in the future (since that may be a behavioral change. Same goes for the case in `StringProtocol+Essentials.swift` as well

Reply:
Thank you for the review. This is indeed wrapped by `#if !FOUNDATION_FRAMEWORK` already:


```swift
#if !FOUNDATION_FRAMEWORK
case .isoLatin1:
  ...
case .macOSRoman:
  ...
case .japaneseEUC:
  ...
#endif
```